kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -130,10 +134,16 @@ public Snapshot buildLatestSnapshot(Engine engine)
     public Snapshot getSnapshotAt(
             Engine engine,
             long version) throws TableNotFoundException {
+        SnapshotImpl upperBoundSnapshot = getSnapshotAtInit(engine);
 
         Optional<LogSegment> logSegmentOpt = getLogSegmentForVersion(engine,
-            Optional.empty(), /* startCheckpointOpt */
-            Optional.of(version) /* versionToLoadOpt */);
+                Optional.empty(), /* startCheckpointOpt */
+                Optional.of(version) /* versionToLoadOpt */,
+                upperBoundSnapshot.getCommitCoordinatorClientHandlerOpt(engine),
+                /* commitCoordinatorClientHandlerOpt */
```

Comment:
[nit] /* commitCoordinatorClientHandlerOpt */ should either be on the right of the parameter or above it.